packages/rest/src/lib/utils/utils.ts

```diff
@@ -121,3 +120,28 @@ export async function onRateLimit(manager: RequestManager, rateLimitData: RateLi
 export function calculateUserDefaultAvatarIndex(userId: Snowflake) {
 	return Number(BigInt(userId) >> 22n) % 6;
 }
+
+/**
+ * Sleeps for a given amount of time.
+ *
+ * @param ms - The amount of time (in milliseconds) to sleep for
+ */
+export async function sleep(ms: number): Promise<void> {
+	return new Promise<void>((resolve) => {
+		setTimeout(() => resolve(), ms);
+	});
+}
+
+/**
+ * Verifies that a value is a buffer-like object.
+ *
+ * @param value - The value to check
+ */
+export function isBufferLike(value: unknown): value is ArrayBuffer | Buffer | Uint8Array | Uint8ClampedArray {
+	return (
+		(typeof Buffer !== 'undefined' && Buffer.isBuffer(value)) ||
```

Comment:
One question, do we strictly need it to be a `Buffer`? Is `Uint8Array` (which `Buffer` extends) not enough?

In fact we check this on line 144, so this check is a bit redundant.